src/presentation/emitters/ingest-event-service.ts

```diff
@@ -0,0 +1,32 @@
+import { IngestEventEmitter } from '../interfaces/ingest-event-emitter'
+import { IngestEventObserver } from '../interfaces/ingest-event-observer'
+import { IngestEventBuilder } from '../interfaces/ingest-event-builder'
+import { IngestEvent } from '../value-objects/ingest-event'
+
+export type IngestEventCallback = (ingest: IngestEvent) => void
+
+export class IngestEventService implements IngestEventEmitter, IngestEventObserver {
+  public static instance: IngestEventService
+
+  public static getInstance(ingestEventBuilder: IngestEventBuilder): IngestEventService {
+    this.instance ??= new IngestEventService(ingestEventBuilder)
+    return this.instance
+  }
+
+  private readonly callbacks: IngestEventCallback[] = []
+
+  private constructor(private readonly ingestEventBuilder: IngestEventBuilder) {}
+
+  public emitTestEvent(queueId: string): void {
+    const event: IngestEvent = this.ingestEventBuilder.buildTestEvent(queueId)
+    this.emitIngestEvent(event)
+  }
```

Comment:
```suggestion
  public emitTestEvent(queueId: string): void {
    this.emitIngestEvent(this.ingestEventBuilder.buildTestEvent(queueId))
  }
```

Is it the norm that we define new variables before using them once only? :) I don't think anyone is in doubt here that is an even that is being built and then emitted 😄

Reply:
That depends on the use case. If the variable is created to easen the understanding of the value yes (e.g. a complex regular expression), otherwise no. I have inlined the event building.